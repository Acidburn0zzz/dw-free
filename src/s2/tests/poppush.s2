#-*-perl-*-

layerinfo type = core;
layerinfo "majorversion" = 0;

function builtin string(int i) : string;
function builtin int(string s) : int;

function main ()
{
    var int[] iarray;
    $iarray[0] = 0;
    $iarray[1] = 1;
    $iarray[2] = 2;
    $iarray[3] = 3;
    $iarray[4] = 4;
    $iarray[5] = 5;

    var int poptest = pop $iarray;
    println "Popped = " + string($poptest);
    println "Size now = " + string(size $iarray);

    $poptest = pop $iarray;
    println "Popped = " + string($poptest);
    println "Size now = " + string(size $iarray);

    $poptest = pop $iarray;
    println "Popped = " + string($poptest);
    println "Size now = " + string(size $iarray);

    println "Done.";
}
