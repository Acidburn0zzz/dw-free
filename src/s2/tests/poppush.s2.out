Popped = 5
Size now = 5
Popped = 4
Size now = 4
Popped = 3
Size now = 3
Done.
